libs/utils/analysis/frequency_analysis.py

```diff
@@ -163,6 +163,124 @@ def _dfg_cpu_frequency_transition_rate(self, cpu):
 # Plotting Methods
 ###############################################################################
 
+    def plotPeripheralClock(self, clk, title='Peripheral Frequency'):
+        """
+        Produce graph plotting the frequency of a particular peripheral clock
+
+        :param title: The title for the chart
+        :type  title: str
+
+        :param clk: The clk name to chart
+        :type  clk: str
+
+        :raises: KeyError
+        """
+        if not self._trace.hasEvents('clock_set_rate'):
+            self._log.warning('Events [clock_set_rate] not found, plot DISABLED!')
+            return
+        rate_df = self._dfg_trace_event('clock_set_rate')
+        enable_df = self._dfg_trace_event('clock_enable')
+        disable_df = self._dfg_trace_event('clock_disable')
+
+        rate_df = rate_df[rate_df.clk_name == clk]
+        rate_df['clock_setting'] = rate_df['rate']/1e6
+        rate_df['state'] = -1
+
+        pd.set_option('display.expand_frame_repr', False)
+
+        if not enable_df.empty:
+            enable_df = enable_df[enable_df.clk_name == clk]
+            enable_df['clock_setting'] = 1;
+        if not disable_df.empty:
+            disable_df = disable_df[disable_df.clk_name == clk]
+            disable_df['clock_setting'] = 0;
```

Comment:
Ditto for the warning

Reply:
Adopted changes, although I couldn't reproduce the warning here with my setup (pandas 0.22).